# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libev/realloc.patch
# Copyright (C) 2008 - 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- libev-4.11/ev.c.orig	2012-02-04 20:09:52.000000000 +0100
+++ libev-4.11/ev.c	2012-05-09 14:52:44.986990152 +0200
@@ -1132,7 +1132,7 @@
 }
 
 static void *
-ev_realloc_emul (void *ptr, long size)
+ev_realloc_emul (void *ptr, size_t size)
 {
 #if __GLIBC__
   return realloc (ptr, size);
@@ -1150,10 +1150,10 @@
 #endif
 }
 
-static void *(*alloc)(void *ptr, long size) = ev_realloc_emul;
+static void *(*alloc)(void *ptr, size_t size) = ev_realloc_emul;
 
 void ecb_cold
-ev_set_allocator (void *(*cb)(void *ptr, long size))
+ev_set_allocator (void *(*cb)(void *ptr, size_t size))
 {
   alloc = cb;
 }
--- libev-4.11/ev.h.orig	2012-02-04 19:54:04.000000000 +0100
+++ libev-4.11/ev.h	2012-05-09 14:53:22.823653291 +0200
@@ -530,7 +530,7 @@
  * or take some potentially destructive action.
  * The default is your system realloc function.
  */
-EV_API_DECL void ev_set_allocator (void *(*cb)(void *ptr, long size));
+EV_API_DECL void ev_set_allocator (void *(*cb)(void *ptr, size_t size));
 
 /* set the callback function to call on a
  * retryable syscall error
